### ISSUE TITLE
Coercion to number `+x` broadens numeric literal types to `number`

### ISSUE THREAD

**Issue by pushkine** [microsoft/TypeScript#43586]
TS stable and nightly

[Playground Link](https://www.typescriptlang.org/play?#code/DYUwLgBAZg9jBcEDsEC8yDcBYAUL2Mam+cRA1EtjgUQBRICUVN6tFTJh6Z9TEA9PwgBRAE6iYozuQIYBQsRKl4cYAJ4AHEBADCMEKIDGIADwAVAHxEzEEAA8wIAHYATAM4QnAVwC2AIwMIAH4IGwAyT18A0QhEb38DKmluGgBDDz0DYxN1LRgoaDgLDCA)

```ts
let foo: 7 = 7;

foo = 7;
foo = +7;
foo = (7);
foo = (+7);
foo = +(7); // Error
foo = +foo; // Error
```
```
Type 'number' is not assignable to type '7'.(2322)
```

Expected no errors.

--------

Expected Behavior : the coercion to number `+x` resolves into `typeof x & number` when `typeof x extends number`

```ts
type Coerce<T> = T extends number ? T & number : number;

foo = +foo as Coerce<typeof foo>;
```

Actual Behavior : the coercion `+x` always resolves to `number`
```ts
type Coerce<T> = number;

```

**Comment by RA80533** [comment]
Type literals are only supported for value or object literals. You can examine an even simpler example of what you're trying to achieve:

```ts
// Type 'number' is not assignable to type '2'.(2322)
const sum: 2 = 1 + 1;
```

**Comment by pushkine** [comment]
I should've been clearer, this issue has nothing to do with `x + y` operations, but `+x` number coercions. 

I had assumed I could demonstrate the existence of an issue just by showing a difference in behavior between `+(0)` and `+0`, since the former throws an error whereas the latter works as expected. I rewrote the report with more details. 

I apologize for the confusion, me filing the bug report at 4 am might have something to do with it 😅

I still don't think this is intended behavior however. @andrewbranch ping

**Comment by andrewbranch** [comment]
There’s always some amount of gray area in the argument of “what operations on literals should be understood,” but at the end of the day, it tends not to matter precisely because we’re talking about literal types. If you have asserted that `foo` has the type `7`, you don’t really need a lot of different ways to express the type or value `7`. These arguments rarely generalize to a real problem in expressiveness, because once you generalize, you’re no longer talking about the literal `7`. But if you have a reason this matters in practice, I’m interested to hear it.
